scripts/watchAmplifyFiles.js

```diff
@@ -1,57 +1,101 @@
 /**
  * This script watches for changes in the amplify directory and copies the changed files to the `amplify` Docker container.
+ * It also watches for TypeScript changes in Lambda directories and compiles them.
  */
 /* eslint-disable no-console */
-const { execSync } = require('child_process');
+const { spawn, execSync } = require('child_process');
 const chokidar = require('chokidar');
+const fs = require('fs');
 const path = require('path');
 
-const watchDir = './amplify';
-const targetDir = '/colonyCDapp';
-
-const watcher = chokidar.watch(watchDir, {
-  ignored: /^(?!.*\.env$).*\/\.[^/]+/, // ignore dotfiles except .env
-  ignoreInitial: true,
-});
+// Function to execute shell commands with error handling
+const executeCommand = (command, successMessage, errorMessage) => {
+  try {
+    execSync(command);
+    console.log(successMessage);
+  } catch (err) {
+    console.error(`${errorMessage}: ${err}`);
+  }
+};
 
+// Function to copy changed files to Docker
 const copyToDockerContainer = (filePath) => {
-  // Compute the relative path from the watched directory
+  const targetDir = '/colonyCDapp';
   const relativePath = path.relative(process.cwd(), filePath);
-  // Construct the target path inside the Docker container
   const targetPath = path.join(targetDir, relativePath);
-
-  // Extract the directory and create it in the container (it might exist, but if it doesn't `docker cp` fill error)
   const targetDirPath = path.dirname(targetPath);
+
+  // Create target directory inside the Docker container
   const createDirectoryCommand = `docker exec amplify mkdir -p "${targetDirPath}"`;
-  try {
-    execSync(createDirectoryCommand);
-    console.log(
-      `Directory ${targetDirPath} was created in amplify container (or it existed before).`,
-    );
-  } catch (err) {
-    console.error(`Error creating directory in amplify container: ${err}`);
-    return;
-  }
+  executeCommand(
+    createDirectoryCommand,
+    `Directory ${targetDirPath} created in amplify container.`,
+    `Error creating directory in amplify container`,
+  );
 
+  // Copy the file to the Docker container
   const dockerCopyCommand = `docker cp "${filePath}" "amplify:${targetPath}"`;
-  try {
-    execSync(dockerCopyCommand);
+  executeCommand(
+    dockerCopyCommand,
+    `File ${filePath} copied to amplify container successfully.`,
+    `Error copying file to amplify container`,
+  );
+};
+
+// Set up the file watcher for the amplify directory
+const startFileWatcher = () => {
+  const watchDir = './amplify';
+
+  const watcher = chokidar.watch(watchDir, {
+    ignored: /^(?!.*\.env$).*\/\.[^/]+/, // Ignore dotfiles except .env
```

Comment:
Can we ignore files inside of the node_modules folder here too if it is only a small addition? 👀

Reply:
Ah yes thanks @iamsamgibbs I forgot about this suggestion of yours!